main.nf

```diff
@@ -0,0 +1,932 @@
+        --stratificationModule Filter \\
+        -l INFO
+    """
+}
+
+
+/*
+ * Step 18 - Generate Software Versions Map
+ * 
+*/
+software_versions = [
+  'FastQC': null, 'Trim Galore!': null, 'BWA': null, 'Picard MarkDuplicates': null, 'GATK': null,
+  'SNPEff': null, 'QualiMap': null, 'Nextflow': "v$workflow.nextflow.version"
+]
+
+/*
+* Step 19 - Generate a YAML file for software versions in the pipeline
+* This is then parsed by MultiQC and the report feature to produce a final report with the software Versions in the pipeline.
+*/ 
+
+process get_software_versions {
+    cache false
+    executor 'local'
+
+    input:
+    val fastqc from fastqc_stdout.collect()
+    val trim_galore from trimgalore_logs.collect()
+    val bwa from bwa_stdout.collect()
+    val markDuplicates from markDuplicates_stdout.collect()
+    val gatk from gatk_stdout.collect()
+    val qualimap from qualimap_stdout.collect()
+    val snpeff from snpeff_stdout.collect()
+
+    output:
+    file 'software_versions_mqc.yaml' into software_versions_yaml
+
+    exec:
+    software_versions['FastQC'] = fastqc[0].getText().find(/FastQC v(\S+)/) { match, version -> "v$version"}
+    software_versions['Trim Galore!'] = trim_galore[0].getText().find(/Trim Galore version: (\S+)/) {match, version -> "v$version"}
+    software_versions['BWA'] = bwa[0].getText().find(/Version: (\S+)/) {match, version -> "v$version"}
+    software_versions['Picard MarkDuplicates'] = markDuplicates[0].getText().find(/Picard version ([\d\.]+)/) {match, version -> "v$version"}
+    software_versions['GATK'] = gatk[0].getText().find(/GATK version ([\d\.]+)/) {match, version -> "v$version"} 
+    software_versions['QualiMap'] = qualimap[0].getText().find(/QualiMap v.(\S+)/) {match, version -> "v$version"}
+    software_versions['SNPEff'] = snpeff[0].getText().find(/SnpEff (\S+)/) {match, version -> "v$version" }
+
+    def sw_yaml_file = task.workDir.resolve('software_versions_mqc.yaml')
+    sw_yaml_file.text  = """
+    id: 'ngi-exoseq'
+    section_name: 'NGI-ExoSeq Software Versions'
+    section_href: 'https://github.com/SciLifeLab/NGI-ExoSeq'
+    plot_type: 'html'
+    description: 'are collected at run time from the software output.'
+    data: |
+        <dl class=\"dl-horizontal\">
+${software_versions.collect{ k,v -> "            <dt>$k</dt><dd>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</dd>" }.join("\n")}
+        </dl>
+    """.stripIndent()
+}
+
+
+/*
+* Step 20 - Collect metrics, stats and other resources with MultiQC in a single call
+*/ 
+
+process multiqc {
+    tag "$name"
+    publishDir "${params.outdir}/MultiQC", mode: 'copy'
+
+    input:
+    file multiqc_config
+    file (fastqc:'fastqc/*') from fastqc_results.collect()
+    file ('trimgalore/*') from trimgalore_results.collect()
+    file ('picard/*') from markdup_results.collect()
+    file ('snpEff/*') from snpeff_results.collect()
+    file ('gatk_variant_eval/*') from gatk_variant_eval_results.collect()
```

Comment:
I think that there should be several more results channels that we can pipe in here, such as Qualimap..?

Reply:
Yes, I added qualimap and GATK's BaseRecalibration on BAM files. c1d540ebf6aa28467c95bfac78b1b7daa1d0e2ea